kernel/work_q.c

```diff
@@ -125,6 +125,12 @@ int k_delayed_work_cancel(struct k_delayed_work *work)
 	}
 
 	if (k_work_pending(&work->work)) {
+                /*
+                 * [Sanechips] clear bit of work.flags,
+                 * when remove from the queue if already submitted.
+                 */
+		atomic_clear_bit(work->work.flags,K_WORK_STATE_PENDING);
```

Comment:
Seems harmless, but why?  Once it's removed from the queue, it's state isn't inspected by the work_q code until it is added to another queue at some point. At that point it gets the bit set again.  This just seems like wasted instructions.   How do you exercise a real bug?